terraform/provision.tf

```diff
@@ -14,4 +14,8 @@ resource "null_resource" "provision" {
   provisioner "remote-exec" {
     inline = "domain='${local.domain}' bash ~/provisioning/${var.distro}.bash"
   }
+
+  provisioner "remote-exec" {
+    inline = "domain='${local.domain}' bash -c \"echo ${random_string.flag_in_var_tmp.result}\" | tee /var/tmp/FLAG"
```

Comment:
Copy pasta, the `domain='${local.domain}'` part is not needed here.

As terraform's [file provisioner doesn't seem to support setting a file's mode](https://github.com/hashicorp/terraform/issues/8811), creating the file through `remote-exec` seems right. Is it `0600`?

Reply:
Apparently not:
```
$ ssh ubuntu@$IP ls -l /var/tmp/FLAG
-rw-rw-r-- 1 ubuntu ubuntu 33 Apr 15 14:02 /var/tmp/FLAG
```
I will add chmod 0600 and chown root.

Reply:
Branch updated:
```
$ ssh ubuntu@$IP ls -l /var/tmp/FLAG
-rw------- 1 root root 33 Apr 15 14:10 /var/tmp/FLAG
```